### PR TITLE
Adds more a11y use cases

### PR DESCRIPTION
**Pull request by chunhtai**
## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: dev/a11y_assessments/lib/use_cases/badge.dart [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  const MainWidget({super.key});
+
+  @override
+  State<MainWidget> createState() => MainWidgetState();
+}
+
+class MainWidgetState extends State<MainWidget> {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
```
Comment (chunhtai) [inline dev/a11y_assessments/lib/use_cases/badge.dart:31]: Not sure what to do with auto focus in this case.

File: dev/a11y_assessments/lib/use_cases/navigation_bar.dart [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      ),
+      bottomNavigationBar: NavigationBar(
+        onDestinationSelected: (int index) {
+          setState(() {
+            currentPageIndex = index;
+          });
+        },
+        indicatorColor: Colors.amber[800],
+        selectedIndex: currentPageIndex,
+        destinations: const <Widget>[
```
Comment (chunhtai) [inline dev/a11y_assessments/lib/use_cases/navigation_bar.dart:46]: same here, there is no control to autofocus on

File: dev/a11y_assessments/lib/use_cases/badge.dart (hunk 1 above)
Comment (yjbanov) [reply dev/a11y_assessments/lib/use_cases/badge.dart:31]: Is the badge not focusable? It seems like a component that should be interactive.

File: dev/a11y_assessments/lib/use_cases/navigation_bar.dart (hunk 2 above)
Comment (yjbanov) [reply dev/a11y_assessments/lib/use_cases/navigation_bar.dart:46]: Are any of those widgets focusable? Is the issue that they lack the `autofocus` property?

**Review by yjbanov — APPROVED** [review]

File: dev/a11y_assessments/lib/use_cases/navigation_bar.dart (hunk 2 above)
Comment (chunhtai) [reply dev/a11y_assessments/lib/use_cases/navigation_bar.dart:46]: badge by it self doesn't have a gesture recognizer. Even if it has a gesture detector, we still need some widget to possess focusnode.

File: dev/a11y_assessments/lib/use_cases/badge.dart (hunk 1 above)
Comment (chunhtai) [reply dev/a11y_assessments/lib/use_cases/badge.dart:31]: badge by it self doesn't have a gesture recognizer. Even if it has a gesture detector, we still need some widget to possess focusnode.

**Pull request merged**